src/NHibernate/Action/CollectionAction.cs

```diff
@@ -79,6 +85,26 @@ protected internal ISessionImplementor Session
 
 		#region IExecutable Members
 
+		public string[] QueryCacheSpaces
+		{
+			get
+			{
+				if (persister.OwnerEntityPersister is ICacheableEntityPersister cacheablePersister)
```

Comment:
We need to add the `SupportsQueryCache` property also for `ICollectionPersister` as the cache strategy can also be set for each collection. So the property getter should look like:
```
(persister as ICacheableCollectionPersister)?.SupportsQueryCache == false ? null : persister.CollectionSpaces;
```

Reply:
ok

Reply:
ok